### PR TITLE
Fix visibility of public bounties

### PR DESCRIPTION
**Pull request by motechFR**
Fixes visibility of public bounties by passing bounty page data to use pages hook

Fixes public pages to show an error if the public space does not exist

### CODE REVIEW

File: components/share/PublicPage.tsx [hunk 1]
```diff
@@ -48,8 +48,9 @@ export default function PublicPage () {
   const pageIdOrPath = router.query.pageId instanceof Array ? router.query.pageId.join('/') : router.query.pageId as string;
   const dispatch = useAppDispatch();
   const { pages, currentPageId, setCurrentPageId } = usePages();
+  const [loadingSpace, setLoadingSpace] = useState(true);
   const [currentSpace] = useCurrentSpace();
-  const [, setSpaces] = useSpaces();
+  const [_, setSpaces] = useSpaces();
```
Comment (mattcasey) [inline components/share/PublicPage.tsx:53]: minor, but this is a warning by the linter since we never use `_`

Comment (motechFR) [reply components/share/PublicPage.tsx:53]: Oops, added as test, forgot to remove. Doing so now

File: components/share/PublicPage.tsx [hunk 2]
```diff
@@ -61,7 +62,11 @@ export default function PublicPage () {
       const spaceDomain = (router.query.pageId as string[])[0];
       charmClient.getPublicSpaceInfo(spaceDomain).then(space => {
         setSpaces([space]);
-      });
+        setLoadingSpace(false);
+      })
+        .catch(() => {
+          setLoadingSpace(false);
```
Comment (mattcasey) [inline components/share/PublicPage.tsx:68]: this loading var needs to be set after loading regular public pages too, right now they never appear to load

Comment (mattcasey) [reply components/share/PublicPage.tsx:68]: Also, this is a good case for `.finally()`, vs adding it to .then() and .catch()

File: components/share/PublicBountyList.tsx [hunk 3]
```diff
@@ -66,6 +68,21 @@ export default function PublicBountyList () {
     if (space) {
       charmClient.listBounties(space.id, true)
         .then(_bounties => {
+
+          const pagesToSet: Record<string, IPageWithPermissions> = _bounties.reduce((pageMap, bounty) => {
```
Comment (mattcasey) [inline components/share/PublicBountyList.tsx:72]: this approach doesn't work, because usePages() will automatically refresh itself every 5 minutes

**Pull request merged**